terraform/modules/ecs/iam-roles.tf

```diff
@@ -16,7 +16,38 @@ resource aws_iam_role "ecs" {
   )
 }
 
-resource "aws_iam_role_policy_attachment" "ecs" {
+resource "aws_iam_policy" "ecrpolicy" {
+  name        = "ecrpolicy"
+  path        = "/"
+  description = "Policy to access ECR"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ecr:BatchCheckLayerAvailability",
+                "ecr:BatchGetImage",
+                "ecr:GetDownloadUrlForLayer",
+                "ecr:GetAuthorizationToken"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
```

Comment:
Here you can use `jsonencode()` with HCL instead of the raw JSON code. An example can be seen in `modules/ecs/iam-roles.tf` . But this one is not wrong if you prefer like this.

Reply:
> Here you can use `jsonencode()` with HCL instead of the raw JSON code. An example can be seen in `modules/ecs/iam-roles.tf` . But this one is not wrong if you prefer like this.

Thanks for the suggestion, good to learn different ways to do it. I'll have a look and see the other way to do it.